### PR TITLE
BBA Driver Cleanup

### PR DESCRIPTION
**Pull request by QuzarDC**
Another pass of removing magic values and cleaning up the BBA driver.

-Added definitions for the values of the RXCONFIG register.
-Minor other cleanups in header.
-Remove magic values based on register defines in header and memory.h
-Remove some comment clutter.
-Correct G2 lock without unlock if reading a small block.
-Refactor link status IRQ handler function, and removing redundant forced autonegotiation. It is already performed by the same init that installs the IRQ handler anyways.

### CODE REVIEW

File: kernel/arch/dreamcast/hardware/network/broadband_adapter.c [hunk 1]
```diff
@@ -892,8 +877,33 @@ static void bba_rx(void) {
             break;
         }
     }
+}
+
+static void bba_link_change(void) {
+
```
Comment (ljsebald) [inline kernel/arch/dreamcast/hardware/network/broadband_adapter.c:883]:
```suggestion
```

**Review by ljsebald — APPROVED** [review]

File: kernel/arch/dreamcast/hardware/network/broadband_adapter.c [hunk 2]
```diff
@@ -18,31 +18,25 @@
...
 /* Configuration definitions */
 
 #define RTL_MEM                 (0x1840000)
 
-#define RX_NOWRAP               1 /* 1 for no wrapping or 0 to use wrapping mode (Ignored for 64Kb buffer length) */
-#define RX_MAX_DMA_BURST        6 /* 2^(4+n) bytes from 0-6 (16b - 1Kb) or 7 for unlimited */
-#define RX_BUFFER_LEN_SHIFT     1 /* 0 : 8Kb, 1 : 16Kb, 2 : 32Kb, 3 : 64Kb */
-#define RX_FIFO_THRESHOLD       0 /* 2^(4+n) bytes from 0-6 (16b - 1Kb) or 7 for none */
-#define RX_EARLY_THRESHOLD      0 /* Early RX Threshold multiplier n/16 or 0 for none */
+#define RX_BUFFER_SHIFT           1 /* 0 : 8Kb, 1 : 16Kb, 2 : 32Kb, 3 : 64Kb */
```
Comment (gyrovorbis) [inline kernel/arch/dreamcast/hardware/network/broadband_adapter.c:30]:
```suggestion
#define RX_BUFFER_SHIFT          1 /* 0 : 8Kb, 1 : 16Kb, 2 : 32Kb, 3 : 64Kb */
```
I think the alignment is off by 1 character?

File: kernel/arch/dreamcast/hardware/network/broadband_adapter.c [hunk 3]
```diff
@@ -215,11 +215,11 @@ struct {
 static uint32 const rtl_mem = MEM_AREA_P2_BASE + RTL_MEM;
 
 /* TX buffer pointers */
-static uint32 const txdesc[4] = {
-    MEM_AREA_P2_BASE + 0x0000 + RTL_MEM + TX_BUFFER_OFFSET,
-    MEM_AREA_P2_BASE + 0x0800 + RTL_MEM + TX_BUFFER_OFFSET,
-    MEM_AREA_P2_BASE + 0x1000 + RTL_MEM + TX_BUFFER_OFFSET,
-    MEM_AREA_P2_BASE + 0x1800 + RTL_MEM + TX_BUFFER_OFFSET,
+static uint32 const txdesc[TX_NB_BUFFERS] = {
+    MEM_AREA_P2_BASE + (TX_BUFFER_LEN * 1) + RTL_MEM + TX_BUFFER_OFFSET,
```
Comment (andressbarajas) [inline kernel/arch/dreamcast/hardware/network/broadband_adapter.c:219]: It looks like functionality changed here and not just removing magic numbers.

**Review by andressbarajas — COMMENTED** [review]
Functionality change?

**Comment by QuzarDC** [comment]
> @QuzarDC Can you sync this PR to make possible to merge?

Should be ready now. The big reason this one has lingered is that there was a desire to test the changed autonegotiations when starting via dc-load-ip, and I just haven't been able to. Have only tested via serial.

**Comment by QuzarDC** [comment]
@ljsebald I was able to successfully test the changes here against a scenario when loading from dcload-ip (rather than disk or dcload-serial). I was able to load multiple network examples then return to dcload-ip with the connection being passed on correctly each time. 

Pretty sure that confirmation was the one thing holding this back.

**Comment by DC-SWAT** [comment]
I tested in DS with FTP server and this brunch works in same way as KOS master. Download 950 KB/s, upload still very slow.
Also I tested booting from dc-load-ip and IP address gets exactly from dc-load.
So I can confirm that nothing broke.

**Review by DC-SWAT — APPROVED** [review]

**Review by ljsebald — APPROVED** [review]

**Pull request merged**